exporter/exporterhelper/common.go

```diff
@@ -51,37 +100,121 @@ func WithStart(start Start) ExporterOption {
 	}
 }
 
+// WithShutdown overrides the default TimeoutSettings for an exporter.
+// The default TimeoutSettings is 5 seconds.
+func WithTimeout(timeout TimeoutSettings) ExporterOption {
+	return func(o *baseExporter) {
+		o.cfg.TimeoutSettings = timeout
+	}
+}
+
+// WithRetry overrides the default RetrySettings for an exporter.
+// The default RetrySettings is to disable retries.
+func WithRetry(retry RetrySettings) ExporterOption {
+	return func(o *baseExporter) {
+		o.cfg.RetrySettings = retry
+	}
+}
+
+// WithQueued overrides the default QueuedSettings for an exporter.
+// The default QueuedSettings is to disable queueing.
+func WithQueued(queued QueuedSettings) ExporterOption {
+	return func(o *baseExporter) {
+		o.cfg.QueuedSettings = queued
+	}
+}
+
 // internalOptions contains internalOptions concerning how an Exporter is configured.
 type baseExporter struct {
-	exporterFullName string
-	start            Start
-	shutdown         Shutdown
+	cfg          *settings
+	sender       requestSender
+	rSender      *retrySender
+	qSender      *queuedSender
+	start        Start
+	shutdown     Shutdown
+	startOnce    sync.Once
+	shutdownOnce sync.Once
 }
 
 // Construct the internalOptions from multiple ExporterOption.
-func newBaseExporter(exporterFullName string, options ...ExporterOption) baseExporter {
-	be := baseExporter{
-		exporterFullName: exporterFullName,
+func newBaseExporter(cfg configmodels.Exporter, options ...ExporterOption) *baseExporter {
+	be := &baseExporter{
+		cfg: &settings{
+			Exporter:        cfg,
+			TimeoutSettings: CreateDefaultTimeoutSettings(),
+			// TODO: Enable queuing by default (call CreateDefaultQueuedSettings
+			QueuedSettings: QueuedSettings{Disabled: true},
+			// TODO: Enable retry by default (call CreateDefaultRetrySettings)
+			RetrySettings: RetrySettings{Disabled: true},
+		},
 	}
 
 	for _, op := range options {
-		op(&be)
+		op(be)
+	}
+
+	if be.start == nil {
+		be.start = func(ctx context.Context, host component.Host) error { return nil }
 	}
 
+	if be.shutdown == nil {
+		be.shutdown = func(ctx context.Context) error { return nil }
+	}
+
+	be.sender = &timeoutSender{cfg: &be.cfg.TimeoutSettings}
+
+	be.rSender = newRetrySender(&be.cfg.RetrySettings, be.sender)
```

Comment:
In case of queued and retry the ordering is different than retry and timeout or timeout and exporter itself. Decided to unify the queuedRetry so that no more special cases required.